### ISSUE TITLE
Create LibState Controller for Search

### ISSUE THREAD

**Issue by sblatz** [mozilla-mobile/fenix#4399]
We should create a controller for Search

For #4395

See `BrowserToolbarController.kt` for a good example or reach out to @sblatz or @csadilek for more clarification on the separation of responsibilities.

**Comment by dragneelfps** [comment]
So we have to extract all the logic from the methods in `SearchInteractor.kt` into `SearchController.kt`, except from those methods that use `navController` ?

**Comment by sblatz** [comment]
@dragneelfps yeah, basically! The interactor should just be invoking a method on the controller or emitting a telemetry event :)

**Comment by dragneelfps** [comment]
@sblatz I created PR #4673 for this issue. Please look into it.

**Comment by sblatz** [comment]
QA can you please test the functionality of the search screen to confirm no regressions? I manually tested and found no issues but wanted to make sure I wasn't missing anything 😄

**Comment by kbrosnan** [comment]
@sblatz search is a bit overloaded. Does this need testing of just search engines and submissions or search engines, history, bookmarks, etc? i.e. all interactions with the address/search bar

**Comment by AndiAJ** [comment]
Hi @sblatz,  I've checked the following and **haven't encountered any issues** on 1.0.1934 (Build # 1) 📦: 10.0.0, 1c652634c GV: 70.0a1-20190822095439 from 23/8 using a Google Pixel 3a (Android 9):

• Search bar UI
• Search suggestions
• Autocomplete
• Recent searches
• History & bookmarks suggestions management
• Search engines respect the locale
• Search engine can be changed temporarily using Shortcuts
• Default engine can be set from Shortcuts
• Verify the Scan button

Please let us know if I've missed something. 😊
Based on the above, I'll mark this ticket as **verified**.
